Mathlib/Algebra/BigOperators/Associated.lean

```diff
@@ -81,6 +81,35 @@ theorem exists_associated_mem_of_dvd_prod [CancelCommMonoidWithZero α] {p : α}
       exact ⟨q, Multiset.mem_cons.2 (Or.inr hq₁), hq₂⟩
 #align exists_associated_mem_of_dvd_prod exists_associated_mem_of_dvd_prod
 
+open Submonoid in
```

Comment:
There is a notion of “divisor-closed submonoid” in the literature — see https://link.springer.com/article/10.1007/s00233-019-10079-0 where they study the relation between closure and divisor closure.
I don't really know what it is worth, but having it may allow to state this theorem as a kind of `divisor_closure_eq_closure`.

Reply:
Using the article, I added the definition of the divisor closure (`Submonoid.divisor_closure`), of a divisor-closed submonoid (`Submonoid.divisor_closed`), two results about the monoid and the trivial monoid being divisor-closed (`top_divisor_closed` and `bot_divisor_closed`). I think it looks neater now. Thank you! I also proved that `closure { r | Prime r }` is divisor-closed (`Submonoid.closure_divisor_closed`). Should I now rename `Submonoid.closure_divisor_closed` to `divisor_closure_eq_closure` and keep the old name for `Associated_closure_of_mul_mem_closure`?

Reply:
I would say that `Submonoid.divisor_closure_eq_closure` is OK.
For the other one, I have no opinion.

Reply:
The definition of `divisor_closed` in that paper is a bit different: “A submonoid S ⊂ H is said to be divisor-closed if every divisor a ∈ H of some element b ∈ S lies in S.” Does it work for you?

Reply:
I think that the definition of the paper entails that `Submonoid.closure { r : α | Prime r }` is not divisor-closed. Therefore, I tweaked the definition a bit to be able to prove `Submonoid.closure_divisor_closed`.

Reply:
The definition of `divisor_closed` in the paper works for me. I made the corresponding changes to the PR so it only depends on this definition now.